tests/test_plugins/test_plugins.py

```diff
@@ -0,0 +1,273 @@
+import shutil
+import textwrap
+from pathlib import Path
+
+import pytest
+
+from ..utils.commands import capture
+
+plugin_pyproject_template = textwrap.dedent(
+    """
+        [tool.poetry]
+        name = "{plugin_name}"
+        authors = ["ManimCE"]
+        version = "0.1.0"
+        description = ""
+
+        [tool.poetry.dependencies]
+        python = "^3.6"
+
+        [tool.poetry.plugins."manim.plugins"]
+        "{plugin_name}" = "{plugin_entrypoint}"
+
+        [build-system]
+        requires = ["poetry-core>=1.0.0"]
+        build-backend = "poetry.core.masonry.api"
+        """
+)
+
+plugin_init_template = textwrap.dedent(
+    """
+    from manim import *
+    class {class_name}(VMobject):
+        def __init__(self):
+            super().__init__()
+            dot1 = Dot(fill_color=GREEN).shift(LEFT)
+            dot2 = Dot(fill_color=BLUE)
+            dot3 = Dot(fill_color=RED).shift(RIGHT)
+            self.dotgrid = VGroup(dot1, dot2, dot3)
+            self.add(self.dotgrid)
+
+        def update_dot(self):
+            self.dotgrid.become(self.dotgrid.shift(UP))
+    def {function_name}():
+        return [{class_name}]
+    """
+)
+
+cfg_file_contents = textwrap.dedent(
+    """
+        [CLI]
+        plugins = test_plugin
+    """
+)
+
+
+def simple_scenes_path():
+    return str(Path(__file__).parent / "simple_scenes.py")
+
+
+def cfg_file_create(cfg_file_contents, path):
+    file_loc = (path / "manim.cfg").absolute()
+    with open(file_loc, "w") as f:
```

Comment:
Would be nicer to use the `.open` method instead imo
```suggestion
    with file_loc.open("w") as f:
```

Reply:
why though both are the same right?

Reply:
Yeah, I just think the method looks a lot better and fits the OOP-ness of pathlib better

Reply:
Related to the comments, down below. Just keep stylistic choices the same wherever possible. In this case:
`with open()` style is used everywheres, so keep it the same.
@friedkeenan If the style is a concern, I suggest opening a PR to change it everywhere in one go everywhere else.

Reply:
Yeah, could probably try to make the switch from `os.path` to `pathlib` too in that PR. Resolving this then